### PR TITLE
fix: Empty ERC20 balance call

### PR DESCRIPTION
**Pull request by 0xonramp**
Fixes the superfluous `balanceOf` calls. Additional refactoring would be needed to support skipping all balance calls against invalid tokens for a particular chain.

### CODE REVIEW

File: src/app/leverage/provider.tsx [hunk 1]
```diff
@@ -144,6 +144,7 @@ export function LeverageProvider(props: { children: any }) {
   const { balances, forceRefetchBalances } = useBalances(
     address,
     leverageTokenAddresses,
+    chainId !== ARBITRUM.chainId,
```
Comment (0xonramp) [inline src/app/leverage/provider.tsx:147]: Avoid calling `balanceOf` if the chain does not match the tokens that will be passed to it.

Comment (janndriessen) [reply src/app/leverage/provider.tsx:147]: `disabled` is a good addition but what about just passing an empty tokens array for non Arbitrum tokens at the moment? Soon we would need to return an array for Ethereum too here.

File: src/lib/hooks/use-balance.ts [hunk 2]
```diff
@@ -63,12 +63,17 @@ export interface TokenBalance {
...
+  address?: string,
+  tokens?: string[],
+  disabled?: boolean,
+) {
   const publicClient = usePublicClient()
   const [balances, setBalances] = useState<TokenBalance[]>([])
 
   const fetchBalances = useCallback(async () => {
-    if (!address || !publicClient || !tokens || tokens.length === 0) return
+    if (!address || !publicClient || !tokens || tokens.length === 0 || disabled)
```
Comment (janndriessen) [inline src/lib/hooks/use-balance.ts:75]: tokens.length === 0 would already prevent fetching here

**Review by janndriessen — COMMENTED** [review]
nice catch!

File: src/app/leverage/provider.tsx (hunk 1 above)
Comment (0xonramp) [reply src/app/leverage/provider.tsx]: @janndriessen good call, seems like `undefined` will work too, tweaked the implementation to better support other chains

**Pull request merged**